dds/DCPS/Serializer.cpp

```diff
@@ -328,6 +328,7 @@ Serializer::Serializer(ACE_Message_Block* chain, const Encoding& encoding)
   , align_rshift_(0)
   , align_wshift_(0)
   , pos_(0)
+  , construction_status_(ConstructionStatusUndefined)
```

Comment:
```C++
  , construction_status_(ConstructionStatusSuccessful)
  , align_rshift_(0)
  , align_wshift_(0)
  , pos_(0)
```

Correct order. Also I don't think we need a new enum member for this.

Reply:
Wasn't sure what the default should be, since some places seemed to explicitly set success and others check against success. This PR was more of a reminder to myself to ask folks about this tomorrow. :)